### DISCUSSION TITLE
Turnstile click with multiple windows and desktops

### DISCUSSION THREAD

**Discussion by oculunto** [seleniumbase/SeleniumBase#2938] [General]
Hello, 
 Since the latest changes to CF and the new PyAutoGUI solution, I cannot use SB in uc mode with multiple windows and desktops on Windows. The window needs to be focused and it even makes it impossible to be working on something else at the same time ( since the program sends tab keys and enter). 
 The last time we had this same problem, I was thinking about a solution using Pywinauto, that makes it possible to virtually send keys only to the targeted window, without it needing to be on focus.
 The problem is, I need to identify the chrome window to be able to target it. This can be done with the window title or with the PID of the window. Additionally note that many chrome windows are open at the same time, so I cannot just use the title of the window) 
 So my question is: is there a reliable way to get the PID of the created chrome session ?

**Comment by mdmintz** [comment]
Sending keys isn't always enough anymore to bypass CAPTCHAs... the mouse must be used, and that requires the active window to be on top. You already asked about the `PID` here: https://github.com/seleniumbase/SeleniumBase/discussions/2727#discussioncomment-9264459

**Reply by oculunto** [reply]
Hello thank you for your answer. Yeah i did before when the problem araised last time and you suggested the PyAutoGUI solution, I thought that thread was locked and old already and it's now relevant again, sorry if I'm missusing the forum.
I think pywinauto would alow sending mouse mouvements virtually too, but don't quote me on that, that's why I am trying to solve it this way.
Thanks for your work

**Comment by oculunto** [comment]
I found a way to get the PID, working on implementing a pywinauto solution now. 
If anyone is interested on the solution for the PID, I just added a debugging port to each window when initializing chrome (as an option/argument), and then find the PID with psutil and processes named chrome and using that specific port. (Found the solution using chatGPT of all places...)
